src/firmware/main/app/world/firmware_robot.c

```diff
@@ -8,31 +8,45 @@ struct FirmwareRobot
     Dribbler_t* dribbler;
     float (*get_robot_position_x)();
     float (*get_robot_position_y)();
+    float (*get_robot_orientation)();
+    float (*get_robot_velocity_x)();
+    float (*get_robot_velocity_y)();
+    float (*get_robot_velocity_angular)();
+    float (*get_battery_voltage)();
+    ControllerState_t* controller_state;
     Wheel_t* front_right_wheel;
     Wheel_t* front_left_wheel;
     Wheel_t* back_right_wheel;
     Wheel_t* back_left_wheel;
+    RobotConstants_t robot_constants;
 };
 
-FirmwareRobot_t* app_firmware_robot_create(Chicker_t* chicker, Dribbler_t* dribbler,
-
-                                           float (*get_robot_position_x)(),
-                                           float (*get_robot_position_y)(),
-                                           Wheel_t* front_right_wheel,
-                                           Wheel_t* front_left_wheel,
-                                           Wheel_t* back_right_wheel,
-                                           Wheel_t* back_left_wheel)
+FirmwareRobot_t* app_firmware_robot_create(
+    Chicker_t* chicker, Dribbler_t* dribbler, float (*get_robot_position_x)(),
+    float (*get_robot_position_y)(), float (*get_robot_orientation)(),
+    float (*get_robot_velocity_x)(), float (*get_robot_velocity_y)(),
+    float (*get_robot_velocity_angular)(), float (*get_battery_voltage)(),
+    Wheel_t* front_right_wheel, Wheel_t* front_left_wheel, Wheel_t* back_right_wheel,
+    Wheel_t* back_left_wheel, ControllerState_t* controller_state,
+    RobotConstants_t robot_constants)
 {
     FirmwareRobot_t* new_robot = malloc(sizeof(FirmwareRobot_t));
 
-    new_robot->chicker              = chicker;
-    new_robot->dribbler             = dribbler;
-    new_robot->get_robot_position_x = get_robot_position_x;
-    new_robot->get_robot_position_y = get_robot_position_y;
-    new_robot->front_right_wheel    = front_right_wheel;
-    new_robot->front_left_wheel     = front_left_wheel;
-    new_robot->back_right_wheel     = back_right_wheel;
-    new_robot->back_left_wheel      = back_left_wheel;
+    new_robot->chicker                    = chicker;
+    new_robot->dribbler                   = dribbler;
+    new_robot->get_robot_position_x       = get_robot_position_x;
+    new_robot->get_robot_position_y       = get_robot_position_y;
+    new_robot->get_robot_orientation      = get_robot_orientation;
+    new_robot->get_robot_velocity_x       = get_robot_velocity_x;
+    new_robot->get_robot_velocity_y       = get_robot_velocity_y;
```

Comment:
Am I correct in assuming that some of these X/Y variables are different, in the sense that some are robot-relative and some are world-relative?

If so I think we should use different letters for each.

Reply:
All are world relative. There is a comment in the jdoc for `app_firmware_robot_create` jdoc indicating as such, copied that comment over the struct itself as well:
```
    // NOTE: Everything here is in the global field reference frame (ie. 0,0 is the center
    // of the field, 0 degrees is towards the enemy goal) unless otherwise specified.
```